### PR TITLE
[3653] Add documentation for Gantt

### LINKED ISSUE #3653: Add documentation for Gantt
* [X] **I have checked that this feature has not yet been suggested by someone else.**

### PR DESCRIPTION
**Pull request by lfasani**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3653

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [X] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [x] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [x] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/specifier/representation-gantt.adoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+image::images/gantt-overview.png[Gantt representation overview]
+
+## Gantt description
+
+To define how your model is represented with a `Gantt` representation, the specifier needs to create a `GanttDescription` in the `View` model.
+
+image::images/gantt-description-overview.png[Gantt description overview]
+
+* `Name` : A name that helps the specifier to know what the GanttDescription is about
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: extra space before :

File: doc/specifier/representation-gantt.adoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+* `Domain Type`: Type of the object on which it is possible to create the representation 
+* `Precondition Expression`: An expression that determine if it is possible to create the representation
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: `An expression that determineS`

File: doc/specifier/representation-gantt.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+* `Date Rounding Expression`: An expression that returns a string that is a time duration that will define how the date is rounded when changing the date from the Gantt representation.
+The string contains 
+** an integer which is the number of time unit.
+** a character among {D,H,m} corresponding to a Day, Hour or minute time unit.
+Example: to round to half a day, the expression must returns `12H`
+
+## Task description
+
+The `TaskDescription` is used to define the direct tasks of its parent in the tasks hierarchy.
+It can be the roots tasks in the Gantt or the sub-tasks in a task.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: root ?

File: doc/specifier/representation-gantt.adoc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+* `Name` : A name that helps the specifier to know what the TaskDescription is about
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: extra space before :

File: doc/specifier/representation-gantt.adoc [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+* `Semantic Candidates Expression`: An expression that returns one or more semantic elements.
+Each semantic element will be represented by a task.
+** `self` variable is the semantic element associated with the `Gantt`.
+* `Domain Type`: Type of the object used to filter the semantic object represented by a task 
+
+All the following expression are called as many times as the number of semantic elements returned by the `Semantic Candidates Expression`.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: `the following expressionS are`

File: doc/specifier/representation-gantt.adoc [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+`self` variable is the current semantic element.
+
+* `Name Expression`: An expression that defines the name of the task
+* `Description Expression`: An expression that defines the description of the task
+* `Start Time Expression`: An expression that returns a `java.time.Instant` to define the startS date of the task.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: `the start date`

File: doc/specifier/representation-gantt.adoc [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+### Task description as sub task
+
+To define tasks as the second level of the tasks hierarchy, create a `TaskDescription` in the `TaskDescription`.
+
+It is also possible to reuse `TaskDescription` so that it is possible to define a recursive definition.
+Use `Reused Task Element Descriptions` field to select existing `TaskDescription`. 
+
+## Tools
+
+Many tools can be configured for `Lane` and `Card`.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: Aren't lane and card words related to deck ?

File: doc/specifier/representation-gantt.adoc [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+By default, a tool does nothing.
+A tool provides access to variables useful in the context of the tool and executes the operations associated with the tool.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: This sentence is not clear to read...

File: doc/specifier/representation-gantt.adoc [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+Available variables:
+** `self`: The semantic element associated with the selected Task
+* `Delete Task Tool`: This tool defines the behavior when the user deletes a task using the button in the task contextual palette in Gantt. 
+Available variables:
+** `self`: The semantic element associated with the selected Task
+
+image::images/gantt-task-tools.png[Task palette tools]
+
+
+* `Edit Task Tool`: This tool defines the behavior when the user uses the handles to change start or end dates or the progress.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc:95]: Do you confirm that for the moment you can't change the name, description or progress from the UI?

File: doc/specifier/representation-gantt.adoc [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+* `Drop Task Tool`:
+This tool defines the behavior when dragging and dropping, in the table part of Gantt, a task before, after or inside another task.
+Available variables:
+** `sourceObject`: The semantic element associated with the dragged task
+** `targetObject`: The semantic element associated with the target task where the task is dropped.
+** `sourceTask`: The dragged `Task`.
+** `targetTaskOrGantt`: The target where the task is dropped.
+It can be `Gantt` or another `Task`.
+** `indexInTarget`: The index of the dropped task inside the list of the brother tasks where it is dropped.
+Index=0 means first position in the brother tasks.
```
Comment (frouene) [inline doc/specifier/representation-gantt.adoc]: The term `brother tasks` is odd.
I propose:
** `indexInTarget`: The index of the dropped task inside the new list of tasks where it is dropped.
Index=0 means first position.

File: doc/user/representation-gantt.adoc [hunk 11]
```diff
@@ -0,0 +1,101 @@
+# Gantt representation
+
+A `Gantt` representation is made of a hierarchy tasks. Each task can contain sub-tasks.
+
+The left part is a table. each row represents a task. The columns of the table provide information about the task.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: One sentence by line

Comment (frouene) [inline doc/user/representation-gantt.adoc]: Missing capitalization on each

File: doc/user/representation-gantt.adoc [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+image::images/gantt-overview.png[Gantt representation overview]
+
+## Gantt display
+
+There are 3 types of task:
+
+* The basic task
+
+image::images/gantt-task.png[Gantt task]
+A color show the progress value of the task. Fully colored mean 100%.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: `A color showS`

Comment (frouene) [inline doc/user/representation-gantt.adoc]: One sentence by line

File: doc/user/representation-gantt.adoc [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+* The group task that is a task which contains sub-tasks
+
+image::images/gantt-task-group.png[Gantt group]
+A color show the progress value of the task in the bottom part.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: `A color showS`

File: doc/user/representation-gantt.adoc [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+A task can have a list of dependencies to other tasks.
+The dependencies are represented by an arrow between two tasks oriented from the dependency to the dependant task.
+
+image:images/gantt-task-dependencies.png[Task Dependencies]
+
+## Gantt Toolbar
+
+image::images/gantt-toolbar.png[Gantt Toolbar]
+
+* `Toggle full screen mode` : It allows to display the representation in full screen
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: extra space before :

File: doc/user/representation-gantt.adoc [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+* `Zoom level`: Choose the time unit in the Gantt (Hour, Half Day, Day, Two Days, Week, Month, Year)
+* `Zoom in`
+* `Zoom out`
+* `Fit to screen`: Set the zoom level that allows to display all the tasks
+* `Display columns`: Allow to hide all the columns in a click. Using the drop down combo, the users chooses the displayed columns. The displayed columns are persisted in the Gantt representation.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: One sentence by line

Comment (frouene) [inline doc/user/representation-gantt.adoc]: `the user chooses the displayed columns`

File: doc/user/representation-gantt.adoc [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+* To reorder, drop the task before or after another.
+* To change its container, drop the task on another task.
+
+Note that it is not possible to drop a task in its own children hierarchy.
+
+
+## Table
+
+The table is made rows and columns. There is one row per task.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: `is made of rows`

Comment (frouene) [inline doc/user/representation-gantt.adoc]: One sentence per line

File: doc/user/representation-gantt.adoc [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+
+The first column presents the hierarchy of tasks. 
+It shows the name of the task and optionally a icon the expand/collapse the sub-tasks.
```
Comment (frouene) [inline doc/user/representation-gantt.adoc]: `aN icon TO expand/collapse`

**Review by frouene — CHANGES_REQUESTED** [review]

File: doc/specifier/representation-gantt.adoc (hunk 9 above)
Comment (lfasani) [reply doc/specifier/representation-gantt.adoc:95]: Yes you can not change the name and the description from the UI. You can do it for the progress.

File: doc/specifier/representation-gantt.adoc (hunk 8 above)
Comment (lfasani) [reply doc/specifier/representation-gantt.adoc]: I rewrote it

**Review by frouene — APPROVED** [review]

**Pull request merged**